spec/datadog/core/telemetry/client_spec.rb

```diff
@@ -3,8 +3,9 @@
 require 'datadog/core/telemetry/client'
 
 RSpec.describe Datadog::Core::Telemetry::Client do
-  subject(:client) { described_class.new(enabled: enabled) }
+  subject(:client) { described_class.new(enabled: enabled, heartbeat_interval: heartbeat_interval) }
```

Comment:
**Note: This test was actually breaking because the let below was called `heartbeat_interval_seconds` and not `heartbeat_interval`.**

Minor: I suggest propagating the `_seconds` prefix internally as well:

```suggestion
  subject(:client) { described_class.new(enabled: enabled, heartbeat_interval_seconds: heartbeat_interval_seconds) }
```